### ISSUE TITLE
collect virtual machine temperature

### ISSUE THREAD

**Issue by pytimer** [influxdata/telegraf#4400]
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:

`system` inputs plugin add temperature metric.

### Current behavior:

I want to use telegraf collect vm temperature, but i not found inputs plugin to do it.

### Desired behavior:

Telegraf have inputs plugin to do it.

### Use case: [Why is this important (helps with prioritizing requests)]

If telegraf inputs plugin can collect temperature, please tell me. Thanks. :)

**Comment by danielnelson** [comment]
We have the [sensors input](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/sensors) for Linux, but I'm assuming you are interested in this on Windows via gopsutil?  I think this would be good to add, do you want to prepare a PR?

**Comment by pytimer** [comment]
Thanks for your reply.

I read `sensors input`, it use `lm-sensors` to collect sensor metrics, but many linux machine not installed lm-sensors package. 

I already read gopsutil code, linux it read machine file to get [temperature](https://github.com/shirou/gopsutil/blob/master/host/host_linux.go#L594:6), and yesterday i send a [PR](https://github.com/shirou/gopsutil/pull/550) to gopsutil,  use WMI to collect Windows temperature. Is it helpful for Telegraf?

Whether Telegraf use gopsutil to get temperature? If not, you can tell me which plugin can add this feature, i will happly to send a PR to implement it.

**Comment by danielnelson** [comment]
It should be acceptable to add a `temp` plugin that uses this interface from gopsutil.

**Comment by pytimer** [comment]
Add a new `temp` input plugin in `system` package, i understand it right?

If right, which metrics name should be included in this plugin. Do you have any idea?

**Comment by danielnelson** [comment]
> Add a new temp input plugin in system package, i understand it right?

Yes

> If right, which metrics name should be included in this plugin.

Since the current interface from gopsutil is:
```go
type TemperatureStat struct {
	SensorKey   string  `json:"sensorKey"`
	Temperature float64 `json:"sensorTemperature"`
}
```

Something like this should be appropriate:
```
temp,sensor=<sensorKey> temp=<sensorTemperature>
```

**Comment by pytimer** [comment]
Now my PR not merge, if it merged, i want to begin this work. 
Is it ok? :-)

**Comment by danielnelson** [comment]
Yeah, can you add a link to the PR?  I think we will remove the system package soon with [this PR](https://github.com/influxdata/telegraf/pull/4406) so you should probably make your plugin a top level plugin:  `plugins/inputs/temp/temp.go`

**Comment by pytimer** [comment]
> can you add a link to the PR? 

Sorry, i am a beginner for this. I'm not understand what is mean, can you tell me what should i do or explain it detail? Thanks very much. :)

I add new `temp` inputs plugin to `plugins/inputs/all/all.go`, like `_ "github.com/influxdata/telegraf/plugins/inputs/temp"` directly?

**Comment by danielnelson** [comment]
Just paste a URL to your PR in a comment.

> I add new temp inputs plugin to plugins/inputs/all/all.go, like _ "github.com/influxdata/telegraf/plugins/inputs/temp" directly?

:+1:

**Comment by pytimer** [comment]
OK. Thanks.

**Comment by pytimer** [comment]
@danielnelson i send PR #4411 , i hope you can review it if you have time. Thanks.

**Comment by danielnelson** [comment]
Closed by #4411
